### PR TITLE
Issue 29: Added Z algorithm in String 

### PR DESCRIPTION
**Pull request by Rutuja-Kolte**

### CODE REVIEW

File: String/Z-Algorithm.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Python Program for Z Algorithm to find the occurrence of a pattern in a string in linear time
+
+# Given string
+T = "abababab"
+
+# Pattern to be found
+P = "aba"
+
```
Comment (rohanbabbar04) [inline String/Z-Algorithm.py:8]: Use dynamic input

File: String/Z-Algorithm.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          count = count +1
+          print("Pattern found at index ",i - len(P) -1) 
+          
+# To print number of times pattern occurs in string if it exists         
+if(count>0):
+    print("Number of times Pattern occurs in string: ",count);
+else:
+    print("Pattern does not exist in string")
+
+
```
Comment (rohanbabbar04) [inline String/Z-Algorithm.py:59]: Add sample I/O

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]
Define a function and put the logic of the code there, use main for only driver code

**Comment by Joshitha18** [comment]
@Rutuja-Kolte do changes as requested.

File: String/Z-Algorithm.py (hunk 2 above)
Comment (Rutuja-Kolte) [reply String/Z-Algorithm.py:59]: will do so

**Pull request closed without merging**